resources/metadata/sonar.yaml

```diff
@@ -65,13 +68,47 @@ spec:
         resourceRef:
           - name: commonPipelineEnvironment
             param: artifactVersion
+      - name: projectKey
+        type: string
+        description: "The project key identifies the project in SonarQube."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: coverageExclusions
+        type: "[]string"
+        description: "A list of patterns that should be excluded from the coverage scan."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: inferJavaBinaries
+        type: bool
+        description: "Find the location of generated Java class files in all modules
+          and pass the option `sonar.java.binaries to the sonar tool."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: inferJavaLibraries
+        type: bool
+        description: "If the parameter `m2Path` is configured for the step `mavenExecute`
+          in the general section of the configuration, pass it as option `sonar.java.libraries`
+          to the sonar tool."
```

Comment:
Looks like this line is wrapped due to auto formatting. How does this look like in the docs or cli?

Reply:
I am not aware that this changes the appearance in the generated docs or in the CLI. My understanding is that it is a YAML feature that lines can be broken up like that. A single space character will be put between the lines in the resulting string. I've tested with the CLI output and there are already other steps, for example [yaml](https://github.com/SAP/jenkins-library/blob/master/resources/metadata/mavenExecuteIntegration.yaml#L69) and resulting [docs](https://sap.github.io/jenkins-library/steps/mavenExecuteIntegration/#logsuccessfulmaventransfers).